### ISSUE TITLE
[JENKINS-17893] Add option to block parent build when using parameterized trigger as a post-build action

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/parameterized-trigger-plugin#565]
<p>I use parameterized trigger in matrix jobs to trigger a deploy after the job builds. But I only want to trigger the deploy job once for each job run, not once for every slave that the build runs on. To do that, it seems that I have to use the plugin as a post-build action, but then I can't block the parent job until the downstream job finishes. I would really like to be able to do that, AND have the parent job succeed/fail based on the downstream job result.</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/owenmehegan.png" title="owenmehegan's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=owenmehegan">owenmehegan</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-17893" target="_blank">Add option to block parent build when using parameterized trigger as a post-build action</a></i></summary>
<i><ul>
<li><b>assignee</b>: <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ikedam">ikedam</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Major
<li><b>component(s)</b>: parameterized-trigger-plugin
<li><b>resolution</b>: Won't Fix
<li><b>resolved</b>: 2013-11-14T19:00:20+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 2
<li><b>imported</b>: 2025-12-07
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>I use parameterized trigger in matrix jobs to trigger a deploy after the job builds. But I only want to trigger the deploy job once for each job run, not once for every slave that the build runs on. To do that, it seems that I have to use the plugin as a post-build action, but then I can't block the parent job until the downstream job finishes. I would really like to be able to do that, AND have the parent job succeed/fail based on the downstream job result.</p></pre>
</details>
</details>
<ul><li><i>environment</i>: <code>Linux</code></li></ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/owenmehegan.png" title="owenmehegan's avatar" /> owenmehegan:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-17893?focusedId=187256&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-187256">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I noticed you have made a lot of commits to this plugin recently - hoping you would be interested in looking at this <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>
</pre>
</details>

</ul>
</details>

<p>I noticed you have made a lot of commits to this plugin recently - hoping you would be interested in looking at this <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>ikedam:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-17893?focusedId=187268&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-187268">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I see your points.
But Jenkins is not designed to decide build results by results of downstream builds, and I do not plan to add features out of that design of Jenkins.</p>

<p>In that case, people seem to use Promoted Builds plugin (<a href="https://wiki.jenkins-ci.org/display/JENKINS/Promoted+Builds+Plugin" class="external-link" target="_blank" rel="nofollow noopener">https://wiki.jenkins-ci.org/display/JENKINS/Promoted+Builds+Plugin</a>).
For build execution exclusion, you can use Throttle Concurrent Builds plugin (<a href="https://wiki.jenkins-ci.org/display/JENKINS/Throttle+Concurrent+Builds+Plugin" class="external-link" target="_blank" rel="nofollow noopener">https://wiki.jenkins-ci.org/display/JENKINS/Throttle+Concurrent+Builds+Plugin</a>) .
Or, though I haven't ever used it, Multijob plugin may help you (<a href="https://wiki.jenkins-ci.org/display/JENKINS/Multijob+Plugin" class="external-link" target="_blank" rel="nofollow noopener">https://wiki.jenkins-ci.org/display/JENKINS/Multijob+Plugin</a>).</p>

<p>Would you see whether above plugins can meet your requirements, and let me know what they cannot meet?</p>
</pre>
</details>

</ul>
</details>

<p>I see your points.<br/>
But Jenkins is not designed to decide build results by results of downstream builds, and I do not plan to add features out of that design of Jenkins.</p>

<p>In that case, people seem to use Promoted Builds plugin (<a href="https://wiki.jenkins-ci.org/display/JENKINS/Promoted+Builds+Plugin" class="external-link" target="_blank" rel="nofollow noopener">https://wiki.jenkins-ci.org/display/JENKINS/Promoted+Builds+Plugin</a>).<br/>
For build execution exclusion, you can use Throttle Concurrent Builds plugin (<a href="https://wiki.jenkins-ci.org/display/JENKINS/Throttle+Concurrent+Builds+Plugin" class="external-link" target="_blank" rel="nofollow noopener">https://wiki.jenkins-ci.org/display/JENKINS/Throttle+Concurrent+Builds+Plugin</a>) .<br/>
Or, though I haven't ever used it, Multijob plugin may help you (<a href="https://wiki.jenkins-ci.org/display/JENKINS/Multijob+Plugin" class="external-link" target="_blank" rel="nofollow noopener">https://wiki.jenkins-ci.org/display/JENKINS/Multijob+Plugin</a>).</p>

<p>Would you see whether above plugins can meet your requirements, and let me know what they cannot meet?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/owenmehegan.png" title="owenmehegan's avatar" /> owenmehegan:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-17893?focusedId=188780&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-188780">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>OK, I understand what you're saying. Here is an alternative suggestion that would help me: can you add an option so that, when the plugin is used as a build step in a matrix build, it only runs once for the whole job, instead of once per axis of the matrix? That would still allow me to trigger the deploy job only once, and have the build job fail if the deploy job fails.</p>
</pre>
</details>

</ul>
</details>

<p>OK, I understand what you're saying. Here is an alternative suggestion that would help me: can you add an option so that, when the plugin is used as a build step in a matrix build, it only runs once for the whole job, instead of once per axis of the matrix? That would still allow me to trigger the deploy job only once, and have the build job fail if the deploy job fails.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>ikedam:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-17893?focusedId=188803&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-188803">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Only aggregations of publishers can run for whole a matrix project.
It sounds good to extend <a href="https://wiki.jenkins-ci.org/display/JENKINS/Any+Build+Step+Plugin" class="external-link" target="_blank" rel="nofollow noopener">Any Build Step plugin</a> to allow a build step run as a aggregation.</p>
</pre>
</details>

</ul>
</details>

<p>Only aggregations of publishers can run for whole a matrix project.<br/>
It sounds good to extend <a href="https://wiki.jenkins-ci.org/display/JENKINS/Any+Build+Step+Plugin" class="external-link" target="_blank" rel="nofollow noopener">Any Build Step plugin</a> to allow a build step run as a aggregation.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/owenmehegan.png" title="owenmehegan's avatar" /> owenmehegan:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-17893?focusedId=189825&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-189825">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I decided to start using the Build Flow plugin to orchestrate this behavior. I just create a build flow that orchestrates the build, deploy, and integration test run against the deploy environment. People can run the "flow" job and watch the whole process, and if something fails then the "flow" is marked as failed. I think this is cleaner.</p>
</pre>
</details>

</ul>
</details>

<p>I decided to start using the Build Flow plugin to orchestrate this behavior. I just create a build flow that orchestrates the build, deploy, and integration test run against the deploy environment. People can run the "flow" job and watch the whole process, and if something fails then the "flow" is marked as failed. I think this is cleaner.</p>
